horiz_interp/include/horiz_interp_bicubic.inc

```diff
@@ -576,53 +499,53 @@ module horiz_interp_bicubic_mod
       enddo
       return
 !  (c) copr. 1986-92 numerical recipes software -3#(-)f.
-   end subroutine bcucof
+   end subroutine BCUCOF_
 
 !-----------------------------------------------------------------------
 
 !> find the lower neighbour of xf in field xc, return is the index
-    function indl(xc, xf)
-    real, intent(in) :: xc(1:)
-    real, intent(in) :: xf
-    integer             :: indl
+    function INDL_(xc, xf)
+    real(FMS_HI_KIND_), intent(in) :: xc(1:)
+    real(FMS_HI_KIND_), intent(in) :: xf
+    integer             :: INDL_
     integer             :: ii
-       indl = 1
+       INDL_ = 1
        do ii=1, size(xc)
          if(xc(ii).gt.xf) return
-         indl = ii
+         INDL_ = ii
        enddo
-       call mpp_error(FATAL,'Error in indl')
+       call mpp_error(FATAL,'Error in INDL_')
     return
-    end function indl
+    end function INDL_
 
 !-----------------------------------------------------------------------
 
 !> find the upper neighbour of xf in field xc, return is the index
-    function indu(xc, xf)
-    real, intent(in) :: xc(1:)
-    real, intent(in) :: xf
-    integer             :: indu
+    function INDU_(xc, xf)
+    real(FMS_HI_KIND_), intent(in) :: xc(1:)
+    real(FMS_HI_KIND_), intent(in) :: xf
+    integer             :: INDU_
     integer             :: ii
        do ii=1, size(xc)
-         indu = ii
+         INDU_ = ii
          if(xc(ii).gt.xf) return
        enddo
-       call mpp_error(FATAL,'Error in indu')
+       call mpp_error(FATAL,'Error in INDU_')
     return
-    end function indu
+    end function INDU_
 
 !-----------------------------------------------------------------------
 
-    subroutine fill_xy(fi, ics, ice, jcs, jce, mask, maxpass)
+    subroutine FILL_XY_(fi, ics, ice, jcs, jce, mask, maxpass)
       integer, intent(in)        :: ics,ice,jcs,jce
-      real, intent(inout)        :: fi(ics:ice,jcs:jce)
-      real, intent(in), optional :: mask(ics:ice,jcs:jce)
+      real(FMS_HI_KIND_), intent(inout)        :: fi(ics:ice,jcs:jce)
+      real(FMS_HI_KIND_), intent(in), optional :: mask(ics:ice,jcs:jce)
       integer, intent(in)        :: maxpass
-      real                       :: work_old(ics:ice,jcs:jce)
-      real                       :: work_new(ics:ice,jcs:jce)
+      real(FMS_HI_KIND_)                       :: work_old(ics:ice,jcs:jce)
+      real(FMS_HI_KIND_)                       :: work_new(ics:ice,jcs:jce)
       logical :: ready
-      real    :: blank = -1.e30
-      real    :: tavr
+      real(FMS_HI_KIND_)    :: blank = real(-1.e30,FMS_HI_KIND_)
+      real(FMS_HI_KIND_)    :: tavr
       integer :: ipass = 0
```

Comment:
mind as well change this too 
integer:: ipass

and later on, ipass=0

Reply:
added this, also changed `blank` to a parameter since it's only used for value comparison